components/ws-proxy/pkg/proxy/workspacerouter.go

```diff
@@ -95,18 +97,20 @@ func matchWorkspaceHostHeader(wsHostSuffix string, headerProvider hostHeaderProv
 		}
 
 		var workspaceID, workspacePort, foreignOrigin, foreignPath string
-		matches := foreignContentHostR.FindStringSubmatch(hostname)
-		if len(matches) == 0 {
-			matches = foreignContentHost2R.FindStringSubmatch(hostname)
-		}
+		matches := foreignContentHost2R.FindStringSubmatch(hostname)
 		if len(matches) == 2 {
 			foreignOrigin = matches[1]
+			if strings.Contains(req.URL.Path, "/__files__/") {
+				// REMOVE this once all workspaces have updated to new vscode version
+				// Don't match if path contains `/__files__/` (blobserve image separator)
+				return false
+			}
 			matches = foreignContentPathR.FindStringSubmatch(req.URL.Path)
 			if matchPort {
 				if len(matches) < 4 {
 					return false
 				}
-				// https://extensions-foreign.ws-eu10.gitpod.io/3000-coral-dragon-ilr0r6eq/index.html
```

Comment:
What defined before this subdomain `extensions-foreign` and why is now `https://{{uuid}}.`? 🤔

Reply:
I think it was designed like this In the beginning because of theia theia and then in our vscode fork we adopted it, then vscode added a check so it has to be an uuid always